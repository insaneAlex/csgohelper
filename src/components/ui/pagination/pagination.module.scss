@import 'styles/variables.scss';

.list {
  flex-wrap: wrap;
  gap: 5px;
  @include box-container;
}

.item {
  width: 2rem;
  height: 2rem;
}

.link {
  background: $box-element-color;
  border-radius: 5px;
  border: none;
  cursor: pointer;
  height: 100%;
  width: 100%;

  @include body-short-link;
}

.linkActive {
  border: 1px solid $main-font-color;
}
