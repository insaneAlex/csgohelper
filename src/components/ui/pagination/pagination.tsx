import {FC} from 'react';
import classNames from 'classnames';

import styles from './pagination.module.scss';

type Props = {pagesCount: number; currentPage: number; onPageChange: (page: number) => void};

export const Pagination: FC<Props> = ({pagesCount, currentPage, onPageChange}) => {
  if (pagesCount === 1) {
    return null;
  }

  const pages = Array.from({length: pagesCount}, (_, i) => i + 1);

  return (
    <ul className={styles.list}>
      {pages.map((page) => (
        <li key={page} className={styles.item}>
          <button
            className={classNames(styles.link, {[styles.linkActive]: page === currentPage})}
            onClick={() => onPageChange(page)}
          >
            {page}
          </button>
        </li>
      ))}
    </ul>
  );
};
