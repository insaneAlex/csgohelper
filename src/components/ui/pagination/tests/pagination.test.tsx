import {fireEvent, render, screen} from '@testing-library/react';
import {Pagination} from '../pagination';

describe('Pagination', () => {
  const onPageChangeMock = jest.fn();
  const props = {pagesCount: 2, currentPage: 1, onPageChange: onPageChangeMock};
  it('should render correctly', () => {
    const {container} = render(<Pagination {...props} />);
    expect(container).toMatchSnapshot();
  });
  describe('when user clicks on another page', () => {
    it('should call onPageChange', () => {
      render(<Pagination {...props} />);
      const button = screen.getAllByRole('button');
      fireEvent.click(button[button.length - 1]);
      expect(onPageChangeMock).toHaveBeenCalledTimes(1);
    });
  });
  describe('when pages count is 1', () => {
    it("shouldn't render buttons list", () => {
      render(<Pagination {...props} pagesCount={1} />);
      const list = screen.queryByRole('list');
      expect(list).not.toBeInTheDocument();
    });
  });
});
