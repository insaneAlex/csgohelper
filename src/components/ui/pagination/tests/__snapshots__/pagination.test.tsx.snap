// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render correctly 1`] = `
<div>
  <ul
    class="list"
  >
    <li
      class="item"
    >
      <button
        class="link linkActive"
      >
        1
      </button>
    </li>
    <li
      class="item"
    >
      <button
        class="link"
      >
        2
      </button>
    </li>
  </ul>
</div>
`;
